File: js/src/reverse.ts
/**
 * Reverses the characters of a string or sentence.
 *
 * @throws {TypeError} if `value` is null or undefined.
 */
export function reverse(value: string): string {
  if (value === null || value === undefined) {
    throw new TypeError('reverse: value must not be null or undefined');
  }

  return value.split('').reverse().join('');
}

File: dotnet/src/PalindromeChecker/Reverse.cs
using System;
using System.Text;

namespace PalindromeChecker;

/// <summary>
/// Reverses the characters of a string or sentence.
/// </summary>
public class Reverse
{
    /// <summary>
    /// Returns <paramref name="value"/> with its characters in reverse order.
    /// </summary>
    /// <exception cref="ArgumentNullException">
    /// Thrown when <paramref name="value"/> is null.
    /// </exception>
    public string Apply(string value)
    {
        if (value is null)
        {
            throw new ArgumentNullException(nameof(value));
        }

        var chars = value.ToCharArray();
        Array.Reverse(chars);
        return new string(chars);
    }
}
